compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/reference/AnnotationReference.kt

```diff
@@ -165,33 +214,6 @@ public sealed class AnnotationReference {
     }
 
     override fun scopeOrNull(parameterIndex: Int): ClassReference? = scope
-
-    override fun toAnnotationSpec(): AnnotationSpec {
-      return AnnotationSpec
-        .builder(classReference.asClassName())
-        .apply {
-          annotation.allValueArguments.forEach { (name, value) ->
-            when (value) {
-              is KClassValue -> {
-                val className = value.argumentType(module).classDescriptor()
-                  .asClassName()
-                addMember("${name.asString()} = %T::class", className)
-              }
-              is EnumValue -> {
-                val enumMember = MemberName(
-                  enclosingClassName = value.enumClassId.asSingleFqName()
-                    .asClassName(module),
-                  simpleName = value.enumEntryName.asString()
-                )
-                addMember("${name.asString()} = %M", enumMember)
-              }
-              // String, int, long, ... other primitives.
-              else -> addMember("${name.asString()} = $value")
-            }
-          }
-        }
-        .build()
```

Comment:
Note that we have now one unified implementation. Neither of these two psi / descriptor specific implementations were complete. The good test coverage we have really stopped me from introducing regressions.